clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1510,7 +1517,8 @@ void Sema::MarkDevice(void) {
         case attr::Kind::SYCLIntelNumSimdWorkItems:
         case attr::Kind::SYCLIntelMaxGlobalWorkDim:
         case attr::Kind::SYCLIntelMaxWorkGroupSize:
-        case attr::Kind::SYCLIntelNoGlobalWorkOffset: {
+        case attr::Kind::SYCLIntelNoGlobalWorkOffset:
+        case attr::Kind::SYCLSimd: {
```

Comment:
Just one more question. This change makes this attribute propagated from called device function up to the kernel. Imagine the following situation: simd attribute is applied to some device function, and reqd_sub_group_size with value not equal to `1` is applied to some other device function. These functions are called from the same kernel. Which `intel_reqd_sub_group_size` metadata value the kernel will have?

Reply:
it is en error to call functions marked intel_reqd_sub_group_size(n) from SIMD kernels if n != 1

Reply:
I think this should be diagnosed then.

Reply:
Right